src/App.js

```diff
@@ -1,47 +1,172 @@
-import {MoviesList} from './components/movie-list/movie-list';
-import {Cart} from './components/cart-component/cart';
-import {Search} from './components/search-box/search-box';
+import { MoviesList } from './components/movie-list/movie-list';
+import { Cart } from './components/cart-component/cart';
+import { Search } from './components/search-box/search-box';
 import React, { useState, useEffect } from 'react';
 import Layout from './components/layout-component/layout';
+import Pagination from './components/pagination-component/pagination';
 import './App.css'
 
 function App() {
-    const [movies, setMovies] = useState([]);
-    const apiKey = process.env.REACT_APP_API;
+  const [movies, setMovies] = useState([]);
+  const [cartItems, setCartItems] = useState([]);
+  const [totalPrice, setTotalPrice] = useState(0);
 
-    useEffect(() => {
-      fetchData();
-    }, []);
+  const [searchField, setSearchField] = useState('');
+  const [sortType, setSortType] = useState('vote_average');
 
-    function fetchData() {
+  const [totalResults, setTotalResults] = useState(0);
+  const [totalPages, setTotalPages] = useState(0);
+  const [currentPage, setCurrentPage] = useState(1);
 
+  const apiKey = process.env.REACT_APP_API;
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  //---------------FETCH MOVIES START-----------------
+
```

Comment:
In love with spread operators? when you set new data you don't need to use spread operator.